database/gormdb.go

```diff
@@ -375,7 +376,31 @@ func (db *GormDB) CreateSubmission(submission *pb.Submission) error {
 	if assignment+group != 2 {
 		return gorm.ErrRecordNotFound
 	}
-	return db.conn.Create(submission).Error
+
+	// If a submission for the given assignment and student/group already exists, update it.
+	// Otherwise create a new submission record
```

Comment:
This comment refers to the database access below, I think. Currently, it is a bit confusing.

Reply:
Fixed.